### PR TITLE
Test cleanup

### PR DESCRIPTION
**Pull request by frett**
@hlbraddock 

no library functional changes, just changes to the tests to make them more explicit in testing user object equality.

### CODE REVIEW

**Comment by hlbraddock** [comment]
I like this approach, in that we retain the capability to compare an entire object when/where necessary, but can compare specific data elements at our discretion.

**Comment by frett** [comment]
this still doesn't truly address the equals method, it just makes the tests not dependent on it.

the hashCode implementation is even more troublesome because of how it's used in HashSets. I'll explain more in the open issue so the problems aren't lost.

File: ldaptive/src/test/java/org/ccci/idm/user/ldaptive/dao/LdaptiveUserDaoIT.java [hunk 1]
```diff
@@ -386,50 +409,61 @@ public void run() {
...
         this.dao.save(user);
 
-        final User found = this.dao.findByEmail(user.getEmail(), Boolean.FALSE);
-        Assert.assertNotNull(found);
-        Assert.assertTrue(user.equals(found));
+        // make sure we can find it and it's valid
+        final User found = this.dao.findByGuid(user.getGuid(), false);
+        assertNotNull(found);
+        assertEquals(user.getGuid(), found.getGuid());
+        assertEquals(user.getEmail(), found.getEmail());
```
Comment (hlbraddock) [inline ldaptive/src/test/java/org/ccci/idm/user/ldaptive/dao/LdaptiveUserDaoIT.java]: Not wanting to check other staff user data?

Comment (frett) [inline ldaptive/src/test/java/org/ccci/idm/user/ldaptive/dao/LdaptiveUserDaoIT.java]: I can, I was just making sure the object was actually stored. I'll test some of the staff specific fields

Comment (frett) [inline ldaptive/src/test/java/org/ccci/idm/user/ldaptive/dao/LdaptiveUserDaoIT.java]: the test isn't going to be exhaustive, but having a random sampling should reveal sudden major failures, and it's easy to modify if we start seeing weird behavior related to specific attributes.

Comment (hlbraddock) [inline ldaptive/src/test/java/org/ccci/idm/user/ldaptive/dao/LdaptiveUserDaoIT.java]: If you test for proxy address or password history, you'll get exception 
`java.lang.AssertionError: expected: java.util.HashSet<[09a87fa0987sdf7sdf, 897asdf987asdf789asfd]> but was: java.util.Collections$UnmodifiableCollection<[09a87fa0987sdf7sdf, 897asdf987asdf789asfd]>`

I suppose you can create an either create an unmodifiable collection or transform it during comparison.

**Comment by hlbraddock** [comment]
+1

**Pull request merged**